bundles/org.openhab.binding.mecmeter/src/main/java/org/openhab/binding/mecmeter/handler/MecMeterHandler.java

```diff
@@ -0,0 +1,408 @@
+    public void dispose() {
+        super.dispose();
+        logger.debug("removing thing..");
+        if (pollFuture != null) {
+            pollFuture.cancel(true);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        MecMeterDeviceConfiguration config = getConfig().as(MecMeterDeviceConfiguration.class);
+        String configCheck = config.isValid();
+
+        if (configCheck != null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, configCheck);
+            return;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        pollFuture = scheduler.scheduleWithFixedDelay(() -> {
+            refresh();
+        }, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Get new data
+     * Function to save Response of the powermeter
+     */
+    private void updateData() {
+        powerMeterResponse = getRealtimeData();
+    }
+
+    /**
+     * Get new realtime data over the network
+     *
+     * @return MecMeterResponse class where json values "are saved"
+     */
+    private @Nullable MecMeterResponse getRealtimeData() {
+        MecMeterResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        MecMeterDeviceConfiguration config = getConfig().as(MecMeterDeviceConfiguration.class);
+
+        try {
+            String basicAuthentication = "Basic " + Base64.getEncoder()
+                    .encodeToString(new String(USERNAME + ":" + config.password).getBytes(StandardCharsets.ISO_8859_1));
+
+            String location = MecMeterBindingConstants.POWERMETER_DATA_URL.replace("%IP%", config.ip.strip());
+
+            ContentResponse response = httpClient.newRequest(location).method(HttpMethod.GET)
+                    .header(HttpHeader.AUTHORIZATION, basicAuthentication).timeout(API_TIMEOUT, TimeUnit.MILLISECONDS)
+                    .send();
+            if (response.getStatus() != 200) {
+                errorMsg = "Reading meter did not succeed: " + response.getReason();
+                logger.error("Request to meter failed: HTTP {}: {}", response.getStatus(), response.getReason());
+            } else {
+                result = gson.fromJson(response.getContentAsString(), MecMeterResponse.class);
+                if (result == null) {
+                    errorMsg = "no data returned";
+                    logger.error("no data returned from meter at {}", location);
+                } else {
+                    resultOk = true;
+                }
+            }
+        } catch (JsonSyntaxException e) {
+            errorMsg = "Configuration is incorrect";
+            logger.error("Error running power meter request: {}", e.getMessage());
+        } catch (IllegalStateException e) {
+            errorMsg = "Connection failed";
+            logger.error("Error running powermeter request: {}", e.getMessage());
```

Comment:
The error log level used in this method seems a bit too severe. The debug/error logging in this method also isn't that useful. That's because a very similar `errorMsg` is always used to update the Thing status. So it might be better to remove all the debug/error logging. WDYT?

Reply:
No, these cases should indeed only happen if there's a bug in the binding - therefore an error is logged, so that people will report the bug.

Reply:
If they are real bugs than we need to keep them indeed. :-)